edweiss-app/app/(app)/courses/[id]/deck/[deckId]/index.tsx

```diff
@@ -0,0 +1,404 @@
+/**
+ * CardListScreen.tsx
+ * Display a list of cards in a deck
+ * 
+ * @file cardListScreen.tsx
+ * @description Screen to display a list of cards in a deck
+ * @author Tuan Dang Nguyen
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import { ApplicationRoute } from '@/constants/Component';
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import RouteHeader from '@/components/core/header/RouteHeader';
+import Icon from '@/components/core/Icon';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { CardListDisplay } from '@/components/memento/CardListDisplayComponent';
+import CreateDeleteEditCardModal from '@/components/memento/CreateDeleteEditCardModal';
+import { CardModalDisplay } from '@/components/memento/ModalDisplay';
+import { callFunction, CollectionOf } from '@/config/firebase';
+import { iconSizes } from '@/constants/Sizes';
+import { useUser } from '@/contexts/user';
+import { useDynamicDocs } from '@/hooks/firebase/firestore';
+import { useRepository } from '@/hooks/repository';
+import { useStringParameters } from '@/hooks/routeParameters';
+import Memento from '@/model/memento';
+import { AppUser } from '@/model/users';
+import { allowToEditDeck, checkDupplication_EmptyField, selectedCardIndices_play, sortingCards } from '@/utils/memento/utilsFunctions';
+import { BottomSheetModal } from '@gorhom/bottom-sheet';
+import { Redirect, router } from 'expo-router';
+import React, { useRef, useState } from 'react';
+import { Modal } from 'react-native';
+import { DecksRepository } from '../_layout';
+
+/**
+ * CardListScreen
+ * Display a list of cards in a deck
+ * 
+ * @returns {ApplicationRoute} Screen to display a list of cards in a deck
+ */
+const CardListScreen: ApplicationRoute = () => {
```

Comment:
I suppose that this is just a path change and not actually new code, so I only did a quick overview and did not analyse everything in much detail. If I am wrong however, please tell me and I will come back to review it more in depth

Reply:
xDD. Sorry, I didn't finish my description yet so it is hard to know how my code was and how it is now. Anyway, I couldn't stretch how much refactoring I had to do just to integrate Memento into courses. This perhaps is the greatest PR I have ever done ahahahha. 
Just to be immediate (I will update the description of PR later), the major changes are: change the path, change the UI alots, change the way users experience with the module such as a modal swipes up for the users to create, edit, delete, share the card, add a feature to share cards, share deck and for professor to publish the deck to all student, and finally optimize things so that every action (create, edit, delete) is in one file.

So given this, can you check a bit through these modifications?
Thanksssss :DDD